analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -65,12 +65,12 @@ public sealed class CollectionsShouldImplementGenericInterface : SonarDiagnostic
 
                         if (SuggestGenericCollectionType(typeSymbol) is { } suggestedGenericType)
                         {
-                            issues ??= new();
-                            issues.Add(Diagnostic.Create(Rule, typeDeclaration.Identifier.GetLocation(), suggestedGenericType));
+                            issues ??= [];
+                            issues.Add(new(typeDeclaration.Identifier, suggestedGenericType));
                         }
                     }
 
-                    issues?.ForEach(d => c.ReportIssue(d));
+                    issues?.ForEach(x => c.ReportIssue(Rule, x.Identifier, x.SuggestedGenericType));
```

Comment:
This `ForEach` causes more damage than it solves. Let's refactor this a bit more:
* Remove `issues` variable and it's handling
* Remove `ForEach`
* Remove `Issue` record
* store the result of `containingType.Interfaces.Concat([containingType.BaseType]).WhereNotNull()` in an array.
* if all survive the current `return` condition, then `foreach` them. 
* inside foreach: if `SuggestGnericCollectionType` returns something, report directly

Reply:
Done in cec3919b05537bfb314a6a7555879011e0e6ecf9